pelican/tests/test_readers.py

```diff
@@ -22,6 +22,50 @@ def read_file(self, path, **kwargs):
         r = readers.Readers(settings=get_settings(**kwargs))
         return r.read_file(base_path=CONTENT_PATH, path=path)
 
+    def assertDictHasSubset(self, dictionary, subset):
+        for key, value in subset.items():
+            if key in dictionary:
+                real_value = dictionary.get(key)
+                self.assertEqual(
+                    value,
+                    real_value,
+                    "Expected %r to have value %r, but was %r" % (key, value, real_value))
```

Comment:
This needs to be `str("Expected %r to have value %r, but was %r") % (key, value, real_value))` because of the way `repr` works in py2 and py3. Otherwise, in py2, the output of `repr` will be implicitly converted to unicode to apply formatting, which caused the errors you had previously. With the change you made to `URLWrappers.__repr__`, you just made it sure that the `repr` is ascii-only and implicit conversion (using `ascii` codec) succeeds.